src/components/PostCard/PostCard.jsx

```diff
@@ -34,10 +41,23 @@ const PostCard = ({ post }) => {
 
   const dispatch = useDispatch();
   const navigate = useNavigate();
-  const { user, token } = useSelector((state) => state.auth);
+
+  const { user, token, bookmarks } = useSelector((state) => state.auth);
 
   const likedByUser = likedBy.some((like) => like.username === user?.username);
 
+  const likeHandler = async (postId) =>
+    likedByUser
+      ? await dispatch(dislikePost({ postId, token }))
```

Comment:
Here since dispatching an action itself runs async call, I am not sure whether async for likeHandler is required. Also VSCode gives a suggestion that await on line 51 and 52 has no effect.